src/main/java/com/datatorrent/netlet/AbstractClient.java

```diff
@@ -49,7 +49,7 @@
 
   public boolean isConnected()
   {
-    return key.isValid() && ((SocketChannel)key.channel()).isConnected();
+    return key != null && key.isValid() && ((SocketChannel)key.channel()).isConnected();
```

Comment:
There is a test failure in jenkins:
{code}
java.lang.NullPointerException
	at com.datatorrent.netlet.AbstractClient.isConnected(AbstractClient.java:52)
	at com.datatorrent.netlet.DefaultEventLoopTest.testStop(DefaultEventLoopTest.java:249)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
{code}